hop/__main__.py

```diff
@@ -8,22 +8,51 @@
 from . import version
 
 
-def append_subparser(subparser, cmd, func):
+def append_subparser(subparser, cmd, func, formatter=False):
+    """Attach a command to a subparser, generating help from docstrings.
+
+    Args:
+        subparser: An argparse-based subparser to attach a command to.
+        cmd: The command name.
+        func: The function to call for said command.
+        formatter: Whether to apply custom formatting for sub-subparser commands.
+
+    """
     assert func.__doc__, "empty docstring: {}".format(func)
     help_ = func.__doc__.split("\n")[0].lower().strip(".")
     desc = func.__doc__.strip()
 
-    parser = subparser.add_parser(
-        cmd,
-        formatter_class=argparse.RawDescriptionHelpFormatter,
-        help=help_,
-        description=desc,
-    )
+    if formatter:
+        parser = subparser.add_parser(
+            cmd,
+            help=help_,
+            description=desc,
+            formatter_class=_SubcommandHelpFormatter
+        )
+    else:
+        parser = subparser.add_parser(cmd, help=help_, description=desc)
 
     parser.set_defaults(func=func)
+
     return parser
 
 
+class _SubcommandHelpFormatter(argparse.RawDescriptionHelpFormatter):
```

Comment:
What do you think if this class is moved to cli.py? I think it fits more there

Reply:
I agree with that. I found a technical issue with doing that however, which is kinda annoying. The main issue is that adding it to the `cli` module adds in a circular dependency as `__main__.py` imports from all modules defining a `_main()` function while the other modules import from `cli.py` to grab commonly used CLI utilities.

Reply:
hmm I see, what do you think of a utility module?

Reply:
I like it. I added this class and the subparser function to `hop.utils.cli` in the latest batch of commits.